### PR TITLE
Add hard/soft contrast options for gruvbox dark

### PR DESCRIPTION
**Pull request by ianpan870102**
Added contrast options hard/soft for gruvbox dark, fixed documentation typo for gruvbox light, and tweaked a few other minor details (mostly colour improvements and naming refactors.).

### CODE REVIEW

**Comment by ianpan870102** [comment]
This PR fixes #470

File: themes/doom-gruvbox-theme.el [hunk 1]
```diff
@@ -20,66 +20,93 @@ determine the exact padding."
...
+   (base5      '("#7c6f64" "#6f6f6f" "brightblack")) ; bg4
+   (base6      '("#928374" "#909090" "brightblack")) ; gray
+   (base7      '("#d5c4a1" "#cccccc" "brightblack")) ; fg2
+   (base8      '("#fbf1c7" "#fbfbfb" "brightwhite")) ; fg0
+   (fg         '("#ebdbb2" "#dfdfdf" "brightwhite")) ; fg/fg1
+   (fg-alt     '("#d5c4a1" "#cccccc" "brightwhite")) ; fg2
+
+   ;; Standardized official colours from gruvbox
+   (grey       '("#928374" "#909090" "brightblack"))   ; gray
+   (gray       '("#928374" "#909090" "brightblack"))   ; gray
```
Comment (hlissner) [inline themes/doom-gruvbox-theme.el:58]: Can you remove `gray`? Not sure why we need a duplicate; it isn't even used.

Comment (ianpan870102) [reply themes/doom-gruvbox-theme.el:58]: Sure, removed.

**Comment by ianpan870102** [comment]
Here are the before-and-after screenshots:

### Before
before

### After
#### Gruvbox (dark) Hard
hard


#### Gruvbox (dark) Medium
medium

#### Gruvbox (dark) Soft

soft

**Comment by ianpan870102** [comment]
P.s. I fixed-up my commits to keep commits cleaner.

**Comment by ema2159** [comment]
Hey Ian! Looks nice! I'll review them as soon as I find a little bit of time.

**Comment by ianpan870102** [comment]
Thanks man, take your time!
By the way @ema2159  I thought I'd give you a heads-up regarding the refactoring I made, it's because I noticed that the person who originally ported doom-gruvbox introduced a new colour olive that takes the colour of green, and green took the colour of aqua. This was not in-line with the colour definitions of the original gruvbox palette on the github page, and I had a tiny bit of hard time when tweaking the colours. So I decided to help standardise the naming, **without suffering existing declarations**:  those that were defined to be olive are now green, which is a more correct name, yet with the same HEX value. To improve readability for future changes, I've added comments after every HEX value, indicating its original name from the original gruvbox palette, e.g. `bright-aqua, bright-orange, bg0, bg0_h, bg0_s` etc.

On top of that, I've corrected the appearance of trivial UI stuff like line numbers, HTML/web-mode tag bracket face and comments colours.

**Comment by ianpan870102** [comment]
[Note:] This PR now not only fixes #470 but also https://github.com/hlissner/emacs-doom-themes/issues/458

**Comment by ianpan870102** [comment]
@ema2159 @hlissner friendly ping

**Comment by ema2159** [comment]
Hey Ian! It looks pretty nice! The only thing I'm a little bit concerned with is the thing with the standardized names. Although I think having standardized names is great, I'm seeing a lot of duplicate colors, so I would rather put a comment with the official spec name of the color at the right of the doom color name, and add an extra color only if it's not in the doom palette. Otherwise, the changes are great! I think for quite some time we've been missing the opportunity of using custom variables in order to expand the theme catalog even more.

**Comment by ianpan870102** [comment]
> I'm seeing a lot of duplicate colors, so I would rather put a comment with the official spec name of the color at the right of the doom color name, and add an extra color only if it's not in the doom palette.

@ema2159 I get your point. The reason I did so was because I find it sometimes more convenient and 'readable' to have two different variable names referencing to the same colour (i.e. same hex value). 
That said, I'll try to revert some of the **duplicate declarations** and **remove** all **unused** extra faces to make the definitions more concise.

**Comment by ianpan870102** [comment]
I've taken your suggestion into account in [this commit](https://github.com/ianpan870102/emacs-doom-themes/commit/5eb722aafe1ef3a4a0c2abcb62f05085472c4268), ready for your final review :-)

**Comment by ema2159** [comment]
Thanks Ian! Sorry for making you do double the work 😅. I'll merge it soon! Thanks for helping that much!

**Comment by ianpan870102** [comment]
> Thanks Ian! Sorry for making you do double the work

No problem at all, you're too kind. Happy to contribute!

**Comment by hlissner** [comment]
@ianpan870102 Do you use `solaire-mode`? It seems `doom-gruvbox`'s `bg` and `bg-alt` are the wrong way around; with `bg-alt` being brighter than `bg`. Was that intentional?

What I expected:
image

What I see:
image

**Comment by hlissner** [comment]
@ianpan870102 Sorry to cram two questions in one, but I have another one:

A user told me that the color for `which-key-group-description-face` is too low-contrast; difficult to distinguish from other keys. Would you mind if I changed it? And if so, what would you suggest it be changed to? I'm thinking yellow, but I'm not familiar enough with the theme.

**Comment by ianpan870102** [comment]
> It seems doom-gruvbox's bg and bg-alt are the wrong way around; with bg-alt being brighter than bg. Was that intentional?

Hi @hlissner, although I'm not a solaire user, I did take into account the `bg` and `bg-alt` faces when modifying the theme. **It was indeed my intention to make the secondary `bg-alt` appear brighter than the default `bg`**. However, I'm willing to file another PR and make `bg-alt` darker if it better adheres to `doom-themes` design guidelines for the sake of consistency. Let me know what you think!

> Would you mind if I changed it? And if so, what would you suggest it be changed to? I'm thinking yellow

By all means go ahead :) Yellow is a good choice indeed.

**Pull request merged**